src/LenderNft.hs

```diff
@@ -30,53 +30,52 @@ import qualified Ledger.Typed.Scripts     as Scripts
 import           Ledger.Value             as Value
 import qualified PlutusTx
 import           PlutusTx.Prelude         hiding (Semigroup (..), unless)
-import qualified Plutus.V1.Ledger.Scripts as Plutus
-import           Prelude                  (Semigroup (..), Show)
-import qualified Common.Utils             as U
-
-{-# INLINABLE lender #-}
-lender :: TokenName
-lender = TokenName { unTokenName = consByteString 76 emptyByteString }  -- L
 
 {-# INLINABLE mkPolicy #-}
-mkPolicy :: ValidatorHash -> TxOutRef -> Integer -> ScriptContext -> Bool
-mkPolicy vh utxo _ ctx = validate
+mkPolicy :: TxOutRef -> ScriptContext -> Bool
+mkPolicy utxo ctx = case mintedValue of
+    [val] -> validate val
+    _     -> False
   where
-    nftIsSentToCollateralSc :: Bool
-    nftIsSentToCollateralSc = traceIfFalse "minted lender nft is not sent to collateral smart contract" (valueOf (U.valueToSc vh ctx) (ownCurrencySymbol ctx) lender == 1)
+    mintFlattened :: [(CurrencySymbol, TokenName, Integer)]
+    mintFlattened = flattenValue $ txInfoMint (scriptContextTxInfo ctx)
+
+    mintedValue :: [(CurrencySymbol, TokenName, Integer)]
+    mintedValue = filter (\(cs, _tn, _n) -> cs == ownCurrencySymbol ctx) mintFlattened
+
+    calculateTokenNameHash :: BuiltinByteString
+    calculateTokenNameHash = consByteString (txOutRefIdx utxo) ((getTxId . txOutRefId) utxo)
+
+    validateTokenName :: TokenName -> Bool
+    validateTokenName tn = unTokenName tn == calculateTokenNameHash
 
-    validateMint :: Integer -> Bool
-    validateMint amount = U.hasUTxO utxo ctx &&
```

Comment:
You removed `U.hasUTxO` check, please add it back.

Reply:
Added back